initializr-generator/src/main/java/io/spring/initializr/generator/language/AnnotationContainer.java

```diff
@@ -49,12 +50,22 @@ public boolean isEmpty() {
 	}
 
 	/**
-	 * Specify if this container has a an annotation with the specified {@link ClassName}.
+	 * Specify if this container has an annotation with the specified {@link ClassName}.
 	 * @param className the class name of an annotation
 	 * @return {@code true} if the annotation with the specified class name exists
 	 */
 	public boolean has(ClassName className) {
-		return this.annotations.containsKey(className);
+		return this.annotations.containsKey(className.getCanonicalName());
+	}
+
+	/**
+	 * Specify if this instance contains an annotation with the specified name.
+	 * @param name the name of the annotation
+	 * @return {@code true} if an annotation with the specified name is present, otherwise
+	 * {@code false}
+	 */
+	public boolean has(String name) {
```

Comment:
I don't really like the concept of an annotation name. What do you think about deprecating the `add` methods, and add two new methods: `addSingle` and `addRepeatable`.

`addSingle` behaves essentially like `add` but if there is more than one annotation of the given class and a customizer is provided it fails. Also `addSingle` doesn't add a annotation if it's already there.

`addRepeatable` can't customize already existing annotations, it always adds a new one.

Reply:
I thought `AnnotationContainer` needed a key to identify specific annotations when there are multiple ones of the same type.

But you're probably right that the annotation name concept adds unnecessary complexity.

So you're suggesting `addSingle`/`addRepeatable` would be cleaner because most use cases don't need to handle specific instances of repeatable annotations?

Reply:
Actually, `customize` might be a fine name too.
From a `ClassName` perspective, `customize` and `replace` seem to have similar meanings.

Reply:
I arrived at this api, what do you think?

https://github.com/mhalbritter/initializr/blob/mh/repeatable-annotations/initializr-generator/src/main/java/io/spring/initializr/generator/language/AnnotationContainer.java

Reply:
Looks good!
I think this is much cleaner than the previous implementation.

I found that single annotations have customize functionality, but repeatable annotations don't. 
Do you think repeatable annotations should also have customize features like single annotations do?

If so, methods like `customizeAllRepeatable`, `customizeRepeatableIf`, or `removeRepeatableIf` might be helpful for more granular control over repeatable annotations.

What do you think? Should I look into adding these features?

Reply:
Thanks for looking at it. Do you want to take this code into your PR?

I don't think customizing repeatable annotations make sense. We can add this functionality later, if the need arises.

Reply:
Okay, I'll update the PR based on the code you provided and mention you.

Reply:
I've updated the PR with your suggested code changes. Ready for review. @mhalbritter